app/services/cache/MongoCacheClient.scala

```diff
@@ -129,40 +148,45 @@ class MongoCacheClient(appConfig: ApplicationConfig, db: () => DefaultDB, applic
     }
 
     fetchAll(Some(credId)) flatMap { maybeNewCache =>
-
       val cache: Cache = maybeNewCache.getOrElse(Cache(credId, Map.empty))
 
+      val d= cache.data + (key -> jsonData)
+
       val updatedCache: Cache = cache.copy(
         id = credId,
         data = cache.data + (key -> jsonData),
-        lastUpdated = DateTime.now(DateTimeZone.UTC)
+        lastUpdated = LocalDateTime.now(ZoneOffset.UTC)
       )
 
-      val document = Json.toJson(updatedCache)
-      val modifier = BSONDocument("$set" -> document)
-
-      collection.update(ordered = false).one(bsonIdQuery(credId), modifier, upsert = true) map { _ => updatedCache }
+        collection.replaceOne(
+          filter = Filters.equal("_id", credId),
+          replacement = updatedCache,
+          ReplaceOptions().upsert(true)
+        ).toFuture().map { _ => updatedCache}
     }
   }
 
-
   /**
     * Removes the item with the specified key from the cache
     */
   def removeByKey[T](credId: String, key: String): Future[Cache] = {
-
     fetchAll(Some(credId)) flatMap { maybeNewCache =>
       val cache = maybeNewCache.getOrElse(Cache(credId, Map.empty))
 
       val updatedCache = cache.copy(
         data = cache.data - (key),
-        lastUpdated = DateTime.now(DateTimeZone.UTC)
+        lastUpdated = LocalDateTime.now(ZoneOffset.UTC)
       )
 
       val document = Json.toJson(updatedCache)
```

Comment:
⛏️ Unused variable.

Reply:
good spot, 
was left there from the old code 😬